velero-plugins/migcommon/restore.go

```diff
@@ -0,0 +1,51 @@
+package migcommon
+
+import (
+	"errors"
+
+	"github.com/sirupsen/logrus"
+	"github.com/vmware-tanzu/velero/pkg/plugin/velero"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/meta"
+)
+
+// RestorePlugin is a restore item action plugin for Velero.
+type RestorePlugin struct {
+	Log logrus.FieldLogger
+}
+
+// AppliesTo returns a velero.ResourceSelector that applies to everything.
+func (p *RestorePlugin) AppliesTo() (velero.ResourceSelector, error) {
+	return velero.ResourceSelector{}, nil
+}
+
+// Execute sets a custom annotation on the item being restored.
+func (p *RestorePlugin) Execute(input *velero.RestoreItemActionExecuteInput) (*velero.RestoreItemActionExecuteOutput, error) {
+	p.Log.Info("[common-restore] Entering common migration restore plugin")
+
+	metadata, err := meta.Accessor(input.Item)
+	if err != nil {
+		return nil, err
+	}
+	name := metadata.GetName()
+	p.Log.Infof("[common-restore] common migration restore plugin for %s", name)
+
+	// Set migmigraiton label on all resources, except ServiceAccounts
+	switch input.Item.DeepCopyObject().(type) {
+	case *corev1.ServiceAccount:
+		break
+	default:
+		migMigrationLabel, exist := input.Restore.Labels[MigMigrationLabelKey]
+		if !exist {
+			return nil, errors.New("migmigration label is not found on restore")
+		}
+		labels := metadata.GetLabels()
+		if labels == nil {
+			labels = make(map[string]string)
+		}
+		labels[PartOfLabelKey] = migMigrationLabel
```

Comment:
I think this deviates from how correlation labels are managed.
The `part-of` label references the CAM application.  
```
app.kubernetes.io/part-of = openshift-migration
```

The individual _Resource_ labels map to the specific resource that created it.  `Type = UID`
```
migmigration = <UID>
```

Seems like we should transfer them exactly as they are in the `Restore`.

[1] https://github.com/konveyor/mig-controller/blob/master/pkg/apis/migration/v1alpha1/resource.go#L27
[2] https://github.com/konveyor/mig-controller/blob/master/pkg/apis/migration/v1alpha1/labels.go#L12

Reply:
We'll be using the part-of label on the Velero CRs in the plugin at some point to distinguish between CAM migration and backup/restore operations, so using it for something else on other resources would be confusing.

Reply:
Thinking about this a bit more, I don't think we want to use part-of at all on these resources. "part-of" means "this resource belongs to this other thing". i.e. the velero backup and restore belong to (are controlled by) the openshift-migration application. These resources are user applications. They don't belong to the migration tool. As Jeff pointed out, I think we just want to use `labels[MigMigrationLabelKey]` instead.

Reply:
Also we should only set the label on namespaced resources. If metadata.GetNamespace() is empty, then we should skip label setting.

Reply:
> I think this deviates from how correlation labels are managed.
> The `part-of` label references the CAM application.
> 
> ```
> app.kubernetes.io/part-of = openshift-migration
> ```
> 
> The individual _Resource_ labels map to the specific resource that created it. `Type = UID`
> 
> ```
> migmigration = <UID>
> ```
> 
> Seems like we should transfer them exactly as they are in the `Restore`.
> 
> [1] https://github.com/konveyor/mig-controller/blob/master/pkg/apis/migration/v1alpha1/resource.go#L27
> [2] https://github.com/konveyor/mig-controller/blob/master/pkg/apis/migration/v1alpha1/labels.go#L12

@jortel  @sseago 
The problem here, is that this will remove `backup` and `restore` every time by using `correlationLabel`. There is no option to list in `multiple namespaces except openshift-migration`, where the backup and restore would be labelled by the same correlation labels. The same logic is used to remove stage pods, but there is an additional selector - we look for **pods** in **each** of the migration namespace.

```golang
const PartOfLabelKey string = "migration.openshift.io/part-of" // (migmigration UID)
```
is not the `app.kubernetes.io/part-of` annotation from the controller. It will be unique. @sseago If you know a better name for a label, saying "this resource is being migrated" then `part-of: migmigration` I would change it.

Reply:
@sseago Will add a check on `namespaced` resources, thanks.

Reply:
@sseago @jortel What about `migration.openshift.io/created-by: <uid>`?

Reply:
"migrated-by" maybe? I.e. this resource doesn't belong to the migration, and "created" might imply original src-cluster creation. But `migration.openshift.io/migrated-by: <uid>` should be pretty clear.